hazelcast/src/test/java/com/hazelcast/config/NearCacheConfigAccessorTest.java

```diff
@@ -37,4 +39,13 @@ public void testConstructor() {
     public void testInitDefaultMaxSizeForOnHeapMaps_whenNull_thenDoNothing() {
         NearCacheConfigAccessor.initDefaultMaxSizeForOnHeapMaps(null);
     }
+
+    @Test
+    public void testNearcacheConfigEqualBeforeAndAfterEqual() {
+        NearCacheConfig config1 = new NearCacheConfig();
+        NearCacheConfig config2 = new NearCacheConfig();
+        NearCacheConfigAccessor.initDefaultMaxSizeForOnHeapMaps(config1);
+
+        assertEquals(config1, config2);
```

Comment:
This test seems counter intuitive, since we change `config1` and expects it to be equal with a fresh unchanged instance?

Reply:
My purpose is exactly this. A config should holds equal to its initial state. An internal need should not affect the equality. At least, this is one way to solve the initial issue. #17952

Reply:
For the purpose and to have less complicated object behavior, maybe another alternative is to give a copy of `NearCacheConfig` whenever a proxy asked for it. And making `initDefaultMaxSizeForOnHeapMaps` method call responsibility of the proxy.  
But given that the approach here is already working, i'm ok with it.